meteor/server/api/rest/v1/playlists.ts

```diff
@@ -786,6 +786,27 @@ export function registerRoutes(registerRoute: APIRegisterHook<PlaylistsRestAPI>)
 		}
 	)
 
+	registerRoute<{ playlistId: string; sourceLayerIds: string }, never, void>(
+		'delete',
+		'/playlists/:playlistId/sourceLayers/:sourceLayerIds',
```

Comment:
A DELETE on a sourceLayers resource(s) implies that (1) something will be deleted, and (2) it will be the sourceLayers, neither being the case here, because it will in fact result in _pieces_ being _stopped_. IMO this should be a POST to `/playlists/:playlistId/clear-source-layers` with a body containing an array of sourceLayerIds.

Reply:
...and the existing endpoint that clears a single layer should be marked as deprecated.

Reply:
I think it being a DELETE does make sense - re-running the same request will end up with the same result: the content will be cleared from the `sourceLayer` "bucket". It therefore matches a DELETE, even if it's not DELETING a piece but rather a collection of stuff.

One thing about marking stuff as deprecated: we need to make sure to properly version the change. I guess deprecating an endpoint is a breaking change, so we need to properly re-name, redirect, etc. For the new major version.

Reply:
>  I guess deprecating an endpoint is a breaking change, so we need to properly re-name, redirect, etc. For the new major version.

No, deprecating is merely a warning to consumers that this is not recommended and will likely be going away at some point, the old method will still be there and functional, unless we want to skip deprecating it and jump straight to removing it.
Keeping the old implementation of this around has barely any cost, so isn't worth the cost of making this a breaking change

Reply:
I discussed this with @nytamin, and the conclusion was that having this as `/clear-sourcelayers` is clearer about the intent and fits better with the other operations in this api.
So we shall go with that approach instead.